src/components/higlass/GenomicProfilesSubscriber.js

```diff
@@ -0,0 +1,221 @@
+    const referenceTracksHeightSum = sum(referenceTracks.map(t => t.height));
+    const profileTracksHeightSum = height - referenceTracksHeightSum - 10;
+    const profileTrackHeight = profileTracksHeightSum / genomicProfilesAttrs.row_infos.length;
+    const profileTracks = genomicProfilesAttrs.row_infos.map((rowInfo, i) => {
+      // Get the uid for the HiGlass track.
+      const trackUid = rowInfo[profileTrackUidKey];
+      // When profiles correspond to cell sets, the profile UID will be the cell set path array.
+      const isPath = Array.isArray(trackUid);
+      // Get the name for the HiGlass track: try the name key first,
+      // then try the tail of the path, and otherwise the track UID.
+      // eslint-disable-next-line no-nested-ternary
+      const trackName = profileTrackNameKey
+        ? rowInfo[profileTrackNameKey]
+        : (isPath ? trackUid[trackUid.length - 1] : trackUid);
+      // If the uid is a path, then try to get the corresponding cell set's color,
+      // if it is currently selected.
+      const setInSelection = isPath ? cellSetSelection?.find(s => isEqual(s, trackUid)) : false;
+      const setColor = isPath ? cellSetColor?.find(s => isEqual(s.path, trackUid))?.color : null;
+      // Get the track UID as a string before passing to HiGlass.
+      const trackUidString = isPath ? trackUid.join('__') : trackUid;
+      // Create the HiGlass track definition for this profile.
+      const track = {
+        type: 'horizontal-bar',
+        uid: `bar-track-${trackUidString}`,
+        data: {
+          type: 'zarr-multivec',
+          url,
+          row: i,
+        },
+        options: {
+          name: trackName,
+          showMousePosition: true,
+          mousePositionColor: foregroundColor,
+          labelColor: (theme === 'dark' ? 'white' : 'black'),
+          labelBackgroundColor: (theme === 'dark' ? 'black' : 'white'),
+          labelShowAssembly: false,
+        },
+        height: profileTrackHeight,
+      };
+      // Set the track color if it is available.
+      if (setColor && setInSelection) {
+        const c = setColor;
+        track.options.barFillColor = `rgb(${c[0]},${c[1]},${c[2]})`;
+      } else {
+        track.options.barFillColor = dimColor;
+      }
+      return track;
+    });
+
+    // Create the higlass view.
+    // The HiGlassLazy component will fill in the fields 'uid',
+    // 'initialXDomain', and 'initialYDomain'.
+    const hgView = {
+      tracks: {
+        top: [
+          ...referenceTracks,
+          ...profileTracks,
+        ],
+        left: [],
+        center: [],
+        right: [],
+        bottom: [],
+        whole: [],
+        gallery: [],
+      },
+      layout: {
+        w: 12,
+        h: 12,
+        x: 0,
+        y: 0,
+        moved: false,
+        static: false,
+      },
+    };
+    return hgView;
+  }, [genomicProfilesAttrs, urls, theme, height, profileTrackUidKey,
+    profileTrackNameKey, cellSetSelection, cellSetColor]);
```

Comment:
Just out of curiosity, but why `useMemo`?

Reply:
I guess just in case this component is re-rendered many times due to other coordination values updating that would not have an effect on the generated view config

Reply:
But probably not a very expensive function to execute either way